lua_modules/ds18b20/ds18b20.lua

```diff
@@ -41,7 +44,49 @@ function setup(dq)
   if(pin == nil) then
     pin = defaultPin
   end
-  ow.setup(pin)
+  ow.setup(pin) 
+end
+temp_result = 0
+
+function callBack(addr, func)
+  -- reset last temp
+  temp_result = 0
+  print(addr:byte(1,9))
+  if checkAddrCrc(addr) then
+    if ((addr:byte(1) == 0x10) or (addr:byte(1) == 0x28)) then
+      -- print("Device is a DS18S20 family device.")
+      parasite = parasitePowered(addr)
+      ow.reset(pin)
+      ow.select(pin, addr)
+      ow.write(pin, 0x44, 1)
+      if parasite then
+        -- Parasite powered, need to wait ~750ms
+        print ("Using parasite power wait")
+        -- TODO: use different wait than delay
+        tmr.alarm(1, 750, 0, function () processRead(addr, func) end)
+      else
+        -- external powered, waiting for finish
+        print ("Using external powered wait")
+        repeat
+          local res = ow.read(pin)
+          tmr.wdclr()
+        until ( res ~= 0 )
+        processRead(addr, func)
+      end
+    end
+  end
+end
+
+function processRead(addr, callback_func)
+  data = readScratchpad(addr)
+  if (checkDataCrc(data)) then
+    temp_result = convertTemperature(data, addr, unit)
+  else
+    temp_result = "Data CRC is not valid"
+  end
+  tmr.wdclr()
```

Comment:
I need do some extra branch for it, first I wanted just fix that bug with parasite powered DS when you do not have whenever it is finished or not, so need wait 750ms and quick solution (even bad) was delay, now trying refactor to have more clean solution